src/vma/sock/sock-redirect.cpp

```diff
@@ -1235,6 +1236,53 @@ int fcntl(int __fd, int __cmd, ...)
 	return res;
 }
 
+/* Do the file control operation described by CMD on FD.
+   The remaining arguments are interpreted depending on CMD.
+
+   This function is a cancellation point and therefore not marked with
+   __THROW.
+   NOTE: VMA throw will never occur during handling of any command.
+   VMA will only throw in case VMA doesn't know to handle a command and the
+   user requested explicitly that VMA will throw an exception in such a case
+   by setting VMA_EXCEPTION_HANDLING accordingly (see README.txt)
+   */
+
+extern "C"
+int fcntl64(int __fd, int __cmd, ...)
+{
+	srdr_logfunc_entry("fd=%d, cmd=%d", __fd, __cmd);
+
+	int res = -1;
+	va_list va;
+	va_start(va, __cmd);
+	unsigned long int arg = va_arg(va, unsigned long int);
+	va_end(va);
+
+	int ret = 0;
+	socket_fd_api* p_socket_object = NULL;
+	p_socket_object = fd_collection_get_sockfd(__fd);
+	if (p_socket_object) {
+		VERIFY_PASSTROUGH_CHANGED(res, p_socket_object->fcntl(__cmd, arg));
```

Comment:
such approach does not work for `fcntl64` as far that original `fcntl` can be called instead of `fcntl64` at https://github.com/Mellanox/libvma/blob/master/src/vma/sock/sockinfo.cpp#L193
Pay attention that on abstract method as https://github.com/Mellanox/libvma/blob/master/src/vma/sock/socket_fd_api.h#L164